### PR TITLE
feat: add sso in authproc

### PR DESCRIPTION
**Pull request by lukasmatusiewicz**
+ formatting
+ back to php-client as a submodule

### CODE REVIEW

**Comment by lukasmatusiewicz** [comment]
sso for authsource comes with another commit in this PR.

File: lib/Auth/Process/PrivacyideaAuthProc.php [hunk 1]
```diff
@@ -204,6 +242,22 @@ private function enrollToken($stateID, $username)
...
+    public static function handleLogout()
+    {
+        SimpleSAML_Logger::debug("privacyIDEA: handle logout. Remove SSO data.");
+
+        /*
+         * This method is static and called after login without providing state
+         * and we can't implement separate SSO for different IdP, SP etc.
+         * So we delete single SSO data.
+         */
+        Session::getSessionFromRequest()->deleteData('privacyidea:privacyidea:sso', 'data');
```
Comment (nilsbehlen) [inline lib/Auth/Process/PrivacyideaAuthProc.php:258]: why is this not SimpleSAML_Session?

Comment (lukasmatusiewicz) [reply lib/Auth/Process/PrivacyideaAuthProc.php]: Yeah, your right. Changed in commit #c364017.

Btw. interesting, that this time was no error...

File: templates/loginform.php [hunk 2]
```diff
@@ -201,7 +201,7 @@ class="float-l erroricon" alt="gtk-dialog-error"/>
                                     echo htmlspecialchars($this->t('{privacyidea:privacyidea:scanTokenQR}'));
                                     ?>
                                     <div class="tokenQR">
-                                        <?php echo '<img src="' . $this->data['tokenQR'] . '" />'; ?>
+                                        <?php echo '<img php-client="' . $this->data['tokenQR'] . '" />'; ?>
```
Comment (nilsbehlen) [inline templates/loginform.php:204]: i think this is an rename error?

File: templates/loginform.php [hunk 3]
```diff
@@ -32,7 +32,7 @@
 /*$head = '';
 if ($this->data['u2fSignRequest']) {
     // Add javascript for U2F support before including the header.
-    $head .= '<script type="text/javascript" src="' . htmlspecialchars(SimpleSAML_Module::getModuleUrl('privacyidea/js/u2f-api.js'), ENT_QUOTES) . '"></script>';
+    $head .= '<script type="text/javascript" php-client="' . htmlspecialchars(SimpleSAML_Module::getModuleUrl('privacyidea/js/u2f-api.js'), ENT_QUOTES) . '"></script>';
```
Comment (nilsbehlen) [inline templates/loginform.php:35]: i think this is an rename error?

**Review by nilsbehlen — CHANGES_REQUESTED** [review]

File: templates/loginform.php (hunk 2 above)
Comment (lukasmatusiewicz) [reply templates/loginform.php:204]: This src is html tag's attribute.

Example:
 Girl in a jacket

Comment (lukasmatusiewicz) [reply templates/loginform.php:204]: ` Girl in a jacket `

File: templates/loginform.php (hunk 3 above)
Comment (lukasmatusiewicz) [reply templates/loginform.php:35]: Here also html tag's attribute.

**Pull request merged**